dc/s2n-quic-dc/src/socket/recv/pool.rs

```diff
@@ -238,6 +243,10 @@ impl FreeList for Free {
     #[inline]
     fn free(&self, descriptor: Descriptor) -> Option<Box<dyn 'static + Send>> {
         let mut inner = self.0.lock().unwrap();
+
+        #[cfg(debug_assertions)]
+        assert!(inner.active.remove(&descriptor.as_usize()));
```

Comment:
is there a reason for doing it this way versus using `debug_assert!`?

Reply:
I tried that first but it turns out debug_assert still type checks the arguments with it disabled so you end up with compilation errors about `active` missing in release mode

Reply:
https://doc.rust-lang.org/src/core/macros/mod.rs.html#334